woocommerce/woocommerce
# Fixed Taxonomy Term Limit On Product Collection Filters

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

This pull request refactors the `TaxonomyItem` component. Previously, `useEntityRecords` was used to populate a cache of all of the terms, but without any pagination, this list only contains the first 10 terms. We don't really _need_ this cache since the query for existing terms and search terms will provide all of the information we need to display suggestions and render tokens. With this in mind I have moved towards using `getEntityRecords` directly from the data store. I was also able to reduce the amount of code in the process.

<img width="277" alt="Screenshot 2024-05-03 at 17 58 08" src="https://github.com/woocommerce/woocommerce/assets/6451942/8b392169-0cc1-49de-bd43-c9645e5a2eff">

There are now three different presentations for term names:

* Terms that are missing (have been deleted but not removed from block) are notated as missing.
* Terms with duplicates have the ID appended to differentiate them.
* Terms will otherwise show the name.

Closes #46850.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Add a "Product categories" and "Product tags" filter to a Product Collection block.
2. Perform the search and add a category. Confirm that there is no limit to the number of terms allowed on the site.
3. Create two categories with the same name and add them. Note that when they are both present they both have an ID.
4. Delete a term after saving it to the block. It should show that it's missing in the filter block on next view.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
